opennms-wmi/src/main/java/org/opennms/netmgt/provision/detector/wmi/WmiDetector.java

```diff
@@ -134,7 +135,17 @@ public boolean isServiceDetected(final InetAddress address, final WmiAgentConfig
         return (result != null && result.getResultCode() != WmiResult.RES_STATE_CRIT
                 && result.getResultCode() != WmiResult.RES_STATE_UNKNOWN);
     }
-    
+
+    @Override
```

Comment:
Having to implement these methods and throw `UnsupportedOperationException`s is a little odd. Can we implement them in the base class and only override them when needed instead?